cabal-install/Distribution/Solver/Modular/Tree.hs

```diff
@@ -37,17 +37,17 @@ type Weight = Double
 -- package, flag, and stanza choices control the traversal order.
 -- TODO: The weight type should be changed from [Double] to Double to avoid
 -- giving too much weight to preferences that are applied later.
-data Tree a =
+data Tree a b =
```

Comment:
I wasn't sure how to document the parameters, since they don't really have a meaning and just allow different types of data to be stored on the nodes between tree traversals.  #2917 uses `a` for install plan scores, so the type is `()` before the scoring traversal and a type that contains the score afterwards.  I could say that `a` is the type of extra values stored on `Done` nodes, and `b` is the type of values stored on choice nodes.  Would that be clearer, or were you looking for something else?